File: python/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0
        
        l = 0
        result = 0
        unique_letters = set()

        for r in range(len(s)):
            while s[r] in unique_letters:
                unique_letters.remove(s[l])
                l += 1
            unique_letters.add(s[r])
            result = max(result, r - l + 1)
        return result

File: golang/3.go
package golang
func lengthOfLongestSubstring(s string) int {
    l := 0
    result := 0
    seen := make(map[byte]bool)

    for r:=0; r<len(s); r++ {
        for seen[s[r]] {
            delete(seen, s[l])
            l++
        }
        seen[s[r]] = true

        if r - l + 1 >  result {
            result = r - l + 1
        }
    }
    return result
}